picard/tagger.py

```diff
@@ -512,18 +518,28 @@ def handle_command_quit(self, argstring):
 
     def handle_command_remove(self, argstring):
         for file in self.iter_all_files():
-            if argstring == file.filename:
-                file.remove()
+            if file.filename == argstring:
+                self.remove([file])
                 return
 
     def handle_command_remove_all(self, argstring):
         for file in self.iter_all_files():
-            file.remove()
+            self.remove([file])
+
+    def handle_command_remove_empty(self, argstring):
+        for album in [a for a in self.albums if a.iterfiles()]:
```

Comment:
This code looks incorrect to me:
- Album.iterfiles() return an iterator, to evaluate if it has elements a conversion to a list is somehow needed
- the logic seems to be reversed, we want to remove empty albums/clusters

I think the following code is what we need (untested):

```python
for album in self.albums:
   if not any(album.iterfiles()):
       self.remove_album(album)

for cluster in self.clusters:
    if not any(cluster.iterfiles()):
            self.remove_cluster(cluster)
```